imagery/query.py

```diff
@@ -473,6 +473,164 @@ def query_latest_imagery(
 
   return frames
 
+def query_latest_imagery_with_segment_ids(
+  segment_ids,
+  custom_ids,
+  min_days,
+  authorization,
+  local_dir,
+  mount=None,
+  num_threads=DEFAULT_THREADS,
+  verbose=False,
+  use_cache=True,
+  skip_cached_frames=False,
+):
+  headers = {
+    "content-type": "application/json",
+    "authorization": f'Basic {authorization}',
+  }
+  frames = []
+  pbar = None
+
+  if verbose:
+    pbar = tqdm(total=len(segment_ids))
+
+  threads = min(MAX_API_THREADS, num_threads)
+  executor = concurrent.futures.ThreadPoolExecutor(max_workers=threads)
+  futures = []
+  assert(len(segment_ids) <= MAX_API_THREADS)
```

Comment:
Would add this assert in the argparse -sg arg. You can add a lambda function to the `type` arg in the add_argument call to make this check. That way there is an error message propagated to the user.

Alternatively, add this limit to the arg help message.